### PR TITLE
Fix tests that fail against Traits master

### PR DESCRIPTION
**Pull request by mdickinson**
This PR fixes a couple of tests that fail against the current Traits master, thanks to the fixes in enthought/traits#771.

I've looked at the parts of the actual business logic that deal with list events, and none of them looks as though it would be adversely affected by enthought/traits#771 (indeed, there's at least [one location](https://github.com/enthought/envisage/blob/295b96f9964b15e394772be502ece972f4b7b8cd/envisage/core_plugin.py#L143-L144) that blindly iterates through the items of `event.added`, and that's currently broken under a slice assignment on Traits < 6.0; it'll work as expected under Traits 6.0).

Related: enthought/traits#300.

### CODE REVIEW

File: envisage/tests/test_slice.py [hunk 1]
```diff
@@ -89,7 +97,7 @@ def test_insert(self):
     def test_extend(self):
         """ extend """
 
-        self.f.l.append([99, 100])
+        self.f.l.extend([99, 100])
```
Comment (corranwebster) [inline envisage/tests/test_slice.py]: Drive-by fix?

Comment (mdickinson) [reply envisage/tests/test_slice.py]: Well, not *just* a drive-by fix: it's also necessary to make the tests pass. But it seems likely to me that a test called `test_extend` was actually intended to test `extend` rather than `append`.
